File: oop/code/oop/src/main/java/com/scaler/lld/bird/dependency/birds/Bird.java
package com.scaler.lld.bird.dependency.birds;

import com.scaler.lld.bird.dependency.beak.Beak;
import com.scaler.lld.bird.dependency.enums.BirdType;
import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public abstract class Bird {
    
    private Double weight;
    private String colour;
    private String size;
    private String beakType;
    private BirdType type;

    private Beak beak;

    public void eat() {
        System.out.println(type + " is eating!");
    }

    public void sleep() {
        System.out.println(type + " is sleeping!");
    }
}

File: oop/code/python/oop/SOLID/bird/Bird.py
from abc import ABC, abstractmethod
import BirdType

class Bird(ABC):
	def __init__(self, weight: int, colour: str, size: int, beakType : str, birdType : BirdType) -> None:
		self.__weight = weight
		self.__colour = colour
		self.__size = size
		self.__beakType = beakType
		self.__birdType = birdType

	@abstractmethod
	def makeSound(self) -> None:
		...